lib/IRGen/IRGen.cpp

```diff
@@ -384,7 +384,12 @@ static bool needsRecompile(StringRef OutputFilename, ArrayRef<uint8_t> HashData,
   // Search for the section which holds the hash.
   for (auto &Section : ObjectFile->sections()) {
     StringRef SectionName;
-    Section.getName(SectionName);
+    if (auto nameOrErr = Section.getName())
+      SectionName = *nameOrErr;
+    else {
```

Comment:
Could you please use braces on both sides?  Its harder to read if only one side uses braces.